### ISSUE TITLE
[BUG] Attempting to open Rancher GUI crashes server.

### ISSUE THREAD

**Issue by trunksbomb** [GabrielOlvH/Industrial-Revolution#577]
**Description:**
- Attempting to open Rancher GUI crashes server.

**Steps to Reproduce:**
- Attempt to open Rancher GUI. Tested with Rancher MK1 and MK2 after freshly placing them in the world. Assuming it happens with MK3 and MK4. Other Indrev machines I'm using in my world don't have this issue.

**Technical Information:**
 - Minecraft: 1.20.1
 - Fabric API: 0.15.7
 - Industrial Revolution: 1.16.5-BETA-Hotfix
 - Modpack: All of Fabric 7 2.0.0

**Logs:**
https://pastebin.com/g3uGCddJ

**Details:**

**Comment by broman** [comment]
Can reproduce on same versions

**Comment by sigmasoldi3r** [comment]
Same here, I'll paste my crash dump to see if helps.

[crash-2025-02-23_20.10.52-server.txt](https://github.com/user-attachments/files/18932760/crash-2025-02-23_20.10.52-server.txt)

Seems related to the GUI of the rancher.

**Comment by NolanHewitt** [comment]
Why do you use 1.16.5 instead of 1.16.7?

**Comment by broman** [comment]
Yeah i’m pretty sure this has been fixed in future versions, try upgrading
